agent/metrics_test.go

```diff
@@ -0,0 +1,51 @@
+package agent
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+)
+
+// TestMetrics_ConsulAutopilotHealthy_Prometheus adds testing around
+// the published autopilot metrics on https://www.consul.io/docs/agent/telemetry#autopilot
+func TestMetrics_ConsulAutopilotHealthy_Prometheus(t *testing.T) {
```

Comment:
I have found that it is generally easier to identify the unit under test when the test name includes the struct and method name. We haven't really documented this, but the pattern I follow is:

```
Test[<Struct>_]<Method>[_<Case>]
```


```suggestion
func TestHTTPHandlers_AgentMetrics_ConsulAutopilotHealthy_Prometheus(t *testing.T) {
```